### ISSUE TITLE
Milestone filter works inconsistently

### ISSUE THREAD

**Issue by wzin** [ZenHubIO/support#864]
**tl;dr** 

I have 16 repos connected to my Zenhub board. Milestone filter doesn't show issues for a repo I'm using unless I go and update the milestone via burndown - then I can see *only* issues for this repo but can't see issues for other repos (using this milestone filter on the board).

**long version**

Among my repos I have: `repoA`, `repoB`, `repoC`, `repoD` in which I have a `Merge`  and `Support/Devops` milestones. Each repo has some issues in every milestone.

- Whenever I go to `repoA` board and filter by `Merge` milestone - I see issues from `repoB`, `repoC` and `repoD` but not from `repoA`. The same thing happens when I open any other repo and try to filter issues by this milestone (`repoA` contains like 15 issues in this milestone). It also does not indicate a due date set on the board milestone filter (while it has a due date set in `repoA`).
- Another case is a `Support/Devops` milestone which exists in all repos. Whenever I filter by this milestone in `repoA` I see all repos' issue but `repoA`'s issues.
- As soon as I go to `repoA`'s burndown chart and update the due date (becasue I'm bugged by not seeing the due date on the board) for `Merge` milestone, I can see`repoA`'s issues and only `reposA`'s issues for this milestone in `Merge` milestone*

*all repos are shown all the time (16/16)

**Comment by azenMatt** [comment]
Hi @wzin 

To confirm the ZenHub logic for this, we match milestones based on them have the same GitHub meta data, so they need: 

- Same name (case sensitive) 
- Same end-date 

right now, I know it is tough to check that these all match, and they need to be updated individually to all be in-sync. 

Would you be able to confirm that this information all matches for the milestones you expect to see together? 


cc @brianleung11-3

**Comment by wzin** [comment]
Let me check that now :)

**Comment by wzin** [comment]
It was the end date that was not matched. Thanks - closing!

**Comment by azenMatt** [comment]
Great :) 

FWIW - we are trying to find a better way to make this clearer!
